apps/heft/src/pluginFramework/HeftLifecycleSession.ts

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import type { AsyncParallelHook } from 'tapable';
+
+import type { IHeftRecordMetricsHookOptions, MetricsCollector } from '../metrics/MetricsCollector';
+import type { ScopedLogger, IScopedLogger } from './logging/ScopedLogger';
+import type { IInternalHeftSessionOptions } from './InternalHeftSession';
+import type { IHeftParameters } from './HeftParameterManager';
+import type { IDeleteOperation } from '../plugins/DeleteFilesPlugin';
+import type { HeftPluginDefinitionBase } from '../configuration/HeftPluginDefinition';
+import type { HeftPluginHost } from './HeftPluginHost';
+
+/**
+ * The lifecycle session is responsible for providing session-specific information to Heft lifecycle
+ * plugins. The session provides access to the hooks that Heft will run as part of lifecycle execution,
+ * as well as access to parameters provided via the CLI. The session is also how you request access to
+ * other lifecycle plugins.
+ *
+ * @public
+ */
+export interface IHeftLifecycleSession {
+  /**
+   * The hooks available to the lifecycle plugin.
+   *
+   * @public
+   */
+  readonly hooks: IHeftLifecycleHooks;
+
+  /**
+   * Contains default parameters provided by Heft, as well as CLI parameters requested by the lifecycle
+   * plugin.
+   *
+   * @public
+   */
+  readonly parameters: IHeftParameters;
+
+  /**
+   * The cache folder for the lifecycle plugin. This folder is unique for each lifecycle plugin,
+   * and will not be cleaned when Heft is run with `--clean`. However, it will be cleaned when
+   * Heft is run with `--clean` and `--clean-cache`.
+   *
+   * @public
+   */
+  readonly cacheFolder: string;
```

Comment:
```suggestion
  readonly cacheFolderPath: string;
```

Maybe? And for `tempFolder`. And probably for other places in the public API too.